CrossSectionAnalysis/cpp/CrossSectionCompute.cxx

```diff
@@ -0,0 +1,340 @@
+/*
+ * This file was created by SET [Surgeon] [Hobbyist developer].
+ */
+#include "CrossSectionCompute.h"
+#include <iostream>
+#include <thread>
+#include <mutex>
+#include <math.h> // sqrt
+#include <vector>
+
+#include <vtkMRMLModelNode.h>
+#include <vtkMRMLMarkupsCurveNode.h>
+#include <vtkMRMLSegmentationNode.h>
+#include <vtkPlane.h>
+#include <vtkCutter.h>
+#include <vtkConnectivityFilter.h>
+#include <vtkContourTriangulator.h>
+#include <vtkMatrix4x4.h> // Should not be required.
+#include <vtkMassProperties.h>
+#include <vtkMath.h> // Pi()
+
+std::mutex mtx;
+
+#define WORKER_MESSAGE(msg) mtx.lock();\
+                            std::cout << msg << std::endl;\
+                            mtx.unlock();
+
+vtkStandardNewMacro(vtkCrossSectionCompute);
+
+void vtkCrossSectionCompute::PrintSelf(ostream& os, vtkIndent indent)
+{
+    vtkObject::PrintSelf(os,indent);
+    
+    os << indent << "numberOfThreads: " << this->numberOfThreads << "\n";
+    os << indent << "inputCenterlineNode: " << this->inputCenterlineNode << "\n";
+    os << indent << "inputSurfaceNode: " << this->inputSurfaceNode << "\n";
+    os << indent << "inputSegmentID: " << this->inputSegmentID << "\n";
+}
+
+// The surface polydata is constant. Create it once only.
+bool vtkCrossSectionCompute::SetInputSurfaceNode(vtkMRMLNode * inputSurface, const std::string& inputSegmentId)
+{
+  inputSurfaceNode = inputSurface;
```

Comment:
Use VTK naming conventions: member variables are capitalized and always prefixed with `this->`. This is _very_ important, as not being able to clearly distinguish between member and local variables can easily lead to coding errors, such as crashes.

Reply:
OK for the 'this->' qualification. 

> member variables are capitalized

You mean, use for example : 

```
vtkSmartPointer<vtkMRMLNode> ANODE;
double AVALUE;
```

instead of 

```
vtkSmartPointer<vtkMRMLNode> aNode;
double AVALUE;
```

Reply:
Member variable name starts with uppercase letter (`this->Something`). Local variable starts with lowercase (`something`). Global constants and enums can be all caps (`SOMETHING`).